tests/test_support.py

```diff
@@ -88,5 +102,23 @@ def test_when_asserting_over_multiple_values_after(self):
         self.assertTrue(attempt_wrapper.was_check_callable_called())
 
 
+class SupportTestCase_overridden_configuration(MigTestCase):
+    def _provide_configuration(self):
+        return 'testconfig'
+
+    def test_provides_the_test_configuration(self):
+        expected_last_dir = 'testconfs-py2' if PY2 else 'testconfs-py3'
+
+        configuration = self.configuration
+
+        # check we have a real config object
+        self.assertIsInstance(configuration, Configuration)
+        # check for having loaded a config file from a test config dir
+        config_file_path_parts = configuration.config_file.split(os.path.sep)
+        config_file_path_parts.pop()  # discard file part
+        config_file_last_dir = config_file_path_parts.pop()
```

Comment:
Minor: a slightly long road to extract the parent dir name, but alright :)
Personally I would probably have gone for
`os.path.basename(os.path.dirname(configuration.config_file))`
or if you're more keen on splitting then
`configuration.config_file.split(os.path.sep)[-2]`
Anyway, I'll leave it at that, but add docstrings to the existing and the new class during merge.